src/api/objects/ObjectAPI.js

```diff
@@ -230,11 +230,19 @@ ObjectAPI.prototype.get = function (identifier, abortSignal) {
  *          each resolving to domain objects matching provided search query and options.
  */
 ObjectAPI.prototype.search = function (query, abortSignal) {
+    return this.queryUsingProvider('searchForObjects', query, abortSignal);
+};
+
+ObjectAPI.prototype.queryUsingProvider = function (queryFunction, query, abortSignal) {
```

Comment:
here we're using a query function instead of just `search` as we need to look for more than just domain object names. we ask each provider if they support the function, and fall back on InMemory if they don't